SimDataFormats/Track/interface/SimTrack.h

```diff
@@ -44,12 +47,31 @@ class SimTrack : public CoreSimTrack {
 
   inline void setVertexIndex(const int v) { ivert = v; }
 
+  void setCrossedBoundaryVars(bool crossedBoundary,
+                              int idAtBoundary,
+                              math::XYZVectorD positionAtBoundary,
+                              math::XYZTLorentzVectorD momentumAtBoundary) {
+    crossedBoundary_ = crossedBoundary;
+    idAtBoundary_ = idAtBoundary;
+    positionAtBoundary_ = positionAtBoundary;
+    momentumAtBoundary_ = momentumAtBoundary;
+  }
+  bool crossedBoundary() const { return crossedBoundary_; }
+  math::XYZVectorD getPositionAtBoundary() const { return positionAtBoundary_; }
```

Comment:
to avoid copies, better to return complex objects by const reference (also in the next accessor below)

Reply:
Fixed